python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -379,12 +378,185 @@ def _find_schedule_time(
+    timezone, and repeat, returning the first time that is later than the passed in timestamp.
+    """
+    start_datetime = pendulum.from_timestamp(start_timestamp, tz=timezone_str)
+
+    # Create a naive (timezone-free) datetime to pass into croniter
+    naive_time = datetime.datetime(
+        year=start_datetime.year,
+        month=start_datetime.month,
+        day=start_datetime.day,
+        hour=start_datetime.hour,
+        minute=start_datetime.minute,
+        second=start_datetime.second,
+        microsecond=start_datetime.microsecond,
+    )
+
+    # Go back an hour to ensure that we consider the full set of possible candidates (otherwise
+    # we might fail to properly consider a time that happens twice during a fall DST transition)
+    if ascending:
+        naive_time = naive_time - datetime.timedelta(hours=1)
+    else:
+        naive_time = naive_time + datetime.timedelta(hours=1)
+
+    cron_iter = CroniterShim(cron_string, naive_time)
+
+    # hourly schedules handle DST transitions differently: they skip times that don't exist
+    # entirely and just move on to the next matching time (instead of returning
+    # the end time of the non-existant interval), and when there are two times that match the cron
+    # string, they return both instead of picking the latter time.
+    cron_parts, nth_weekday_of_month, *_ = CroniterShim.expand(cron_string)
+    repeats_every_hour = len(cron_parts[1]) == 1 and cron_parts[1][0] == "*"
+
+    while True:
+        if ascending:
+            next_date = cron_iter.get_next(datetime.datetime)
+        else:
+            next_date = cron_iter.get_prev(datetime.datetime)
+
+        try:
+            dates_to_consider = [
+                create_pendulum_time(
+                    next_date.year,
+                    next_date.month,
+                    next_date.day,
+                    next_date.hour,
+                    next_date.minute,
+                    next_date.second,
+                    next_date.microsecond,
+                    tz=timezone_str,
+                    dst_rule=pendulum.TRANSITION_ERROR,
+                )
+            ]
+        except pendulum.tz.exceptions.NonExistingTime:  # type:ignore
+            if repeats_every_hour:
+                # hourly schedules just move on to the next time
+                dates_to_consider = []
+            else:
+                # other schedules advance to the time at the end of the interval (so that e.g.
+                # a daily schedule doesn't miss an entire day)
+                dates_to_consider = [
+                    create_pendulum_time(
+                        next_date.year,
+                        next_date.month,
+                        next_date.day,
+                        next_date.hour + 1,
+                        0,
+                        0,
+                        0,
+                        tz=timezone_str,
+                        dst_rule=pendulum.TRANSITION_ERROR,
```

Comment:
we expect that moving an hour shouldn't cause an exception to raise a second time i assume?

Reply:
yeah the max time range for ambiguity is an hour